cpp/arcticdb/async/async_store.hpp

```diff
@@ -281,17 +293,19 @@ class AsyncStore : public Store {
     }
 
     std::vector<VariantKey> batch_read_compressed(
-        std::vector<entity::VariantKey> &&keys,
-        std::vector<ReadContinuation> &&continuations,
+        std::vector<entity::VariantKey> &&ks,
+        std::vector<ReadContinuation> &&cs,
         const BatchReadArgs & args) override {
+        auto continuations = std::move(cs);
+        auto keys = std::move(ks);
         util::check(keys.size() == continuations.size(),
                     "keys and continuations must has then same number of elements");
         std::vector<folly::Future<VariantKey>> batch;
         batch.reserve(args.batch_size_);
         std::vector<VariantKey> res;
         res.reserve(keys.size());
         for (std::size_t i = 0; i < keys.size(); ++i) {
-            auto &key = keys[i];
+            const auto &key = keys[i];
```

Comment:
The variable is only used once, so might as well do `std::move(keys[i])` for efficiency?